### PR TITLE
output on activity rectangle hovered

### LINKED ISSUE #222: output on activity rectangle hovered
placeholder to split

### PR DESCRIPTION
**Pull request by umarmohammed**
closes #222

### CODE REVIEW

File: projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts [hunk 1]
```diff
@@ -62,6 +60,8 @@ export class NgxD3timelineComponent
   }
 
   @Output() activityDropped = new EventEmitter<Activity>();
+  @Output() hovered = new EventEmitter<Activity>();
```
Comment (wflemingnz-retired) [inline projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts:63]:
```suggestion
  @Output() activityHovered = new EventEmitter<Activity>();
```

File: projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts [hunk 2]
```diff
@@ -62,6 +60,8 @@ export class NgxD3timelineComponent
...
+  @Output() unhovered = new EventEmitter<Activity>();
```
Comment (wflemingnz-retired) [inline projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts:64]:
```suggestion
  @Output() activityUnhovered = new EventEmitter<Activity>();
```

File: projects/ngx-d3timeline/src/lib/store/state.ts [hunk 3]
```diff
@@ -12,6 +13,7 @@ export interface State {
   lastDraggedActivityId: identifier;
   dragEvent: TimelineDragEvent;
   zoomEvent: any;
+  hoveredEvent: HoverEvent;
```
Comment (wflemingnz-retired) [inline projects/ngx-d3timeline/src/lib/store/state.ts]:
```suggestion
  hoverEvent: HoverEvent;
```
is consistent with zoomEvent and dragEvent

File: projects/ngx-d3timeline/src/lib/hover/hover-event.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { identifier } from '../core/types';
+import {
+  Activity,
+  getActivityFromPositionedActivity
+} from '../activity/activity';
+import { PositionedActivity } from '../activity/positioned-activity';
+
+export enum HoverType {
```
Comment (wflemingnz-retired) [inline projects/ngx-d3timeline/src/lib/hover/hover-event.ts]: maybe `hoverStatus` ? hovered is not really a type of hover

**Review by wflemingnz-retired — CHANGES_REQUESTED** [review]

File: projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts (hunk 1 above)
Comment (umarmohammed) [reply projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts:63]: This event will also have to fire when a resource is hovered

File: projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts (hunk 2 above)
Comment (umarmohammed) [reply projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts:64]: This event will also have to fire when a resource is hovered

File: projects/ngx-d3timeline/src/lib/hover/hover-event.ts (hunk 4 above)
Comment (umarmohammed) [reply projects/ngx-d3timeline/src/lib/hover/hover-event.ts]: How about `HoverAction`?

File: projects/ngx-d3timeline/src/lib/store/state.ts (hunk 3 above)
Comment (umarmohammed) [reply projects/ngx-d3timeline/src/lib/store/state.ts]: done

File: projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts (hunk 1 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts:63]: what will its generic type be then? ie: currently its argument is an Activity, if it handles both will it be Activity | Resource ?

Also we are differentiating hover and unhover, which (arguably) are more closely linked then hovering an Activity and hovering a Resource

File: projects/ngx-d3timeline/src/lib/hover/hover-event.ts (hunk 4 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/hover/hover-event.ts]: yeah thought of that, maybe a bit close to state actions, but I'm fine with it and its way better than HoverType

File: projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts (hunk 1 above)
Comment (umarmohammed) [reply projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts:63]: we don't have to specify the generic type. A resource is always a string so could be Activity | string i.e. object or string.

**Pull request merged**